pkg/osutil/machineid.go

```diff
@@ -0,0 +1,91 @@
+package osutil
+
+import (
+	"bytes"
+	"encoding/xml"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+	"sync"
+
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	machineIDCached string
+	machineIDOnce   sync.Once
+)
+
+func MachineID() string {
+	machineIDOnce.Do(func() {
+		x, err := machineID()
+		if err == nil && x != "" {
+			machineIDCached = x
+			return
+		}
+		logrus.WithError(err).Debug("failed to get machine ID, falling back to use hostname instead")
```

Comment:
I feel slightly uneasy about the fallback, and would rather tell the user to fix the cause why the machine id cannot be determined. Because once they fix it, all the hashes will change.

Reply:
In a container, dbus is almost always missing, so lacking machine ID is completely fine.